src/hotspot/cpu/aarch64/sharedRuntime_aarch64.cpp

```diff
@@ -827,6 +832,11 @@ int SharedRuntime::c_calling_convention(const BasicType *sig_bt,
         if (fp_args < Argument::n_float_register_parameters_c) {
           regs[i].set1(FP_ArgReg[fp_args++]->as_VMReg());
         } else {
+#ifdef __APPLE__
+          // Less-than word types are stored one after another.
+          // The code unable to handle this, bailout.
+          return -1;
+#endif
```

Comment:
This looks like a bug to me. The caller doesn't necessarily check the return value. See CallRuntimeNode::calling_convention.

Reply:
Hello, we have updated PR, now this bailout is used only by the code which can handle it (native wrapper generator), for the rest it will cause guarantee failed if this bailout is triggered

Reply:
This is when passing a float, yes? In the case where we have more float arguments than n_float_register_parameters_c.
I don't understand why you think it's acceptable to bail in this case. Can you explain, please?

Reply:
it's for everything that uses less than 8 bytes on a stack( ints ( 4), shorts(2), bytes(1), floats(4)).
currently native wrapper generation does not support such cases at all, it needs refactoring before this can be implemented.
So when a method has more argument than can be placed in registers, we may have issues. 

So we just bailing out to interpreter in case when a smaller (<=4 b) type is going to be passed thru the stack.

There was attempt to implement handling such cases but currently it requires some hacks (like using some vectors for non-specific task) - https://github.com/openjdk/aarch64-port/pull/3

Reply:
OK. I checked and the Panama preview doesn't support direct native calls for stack arguments, so we're good for now.